src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -1,50 +1,218 @@
+      if (newestInstant == null || u.getCreationTime().compareTo(newestInstant) > 0) {
+        newestInstant = u.getCreationTime();
+        newestUser = u.getName();
+      }
+    }
+
+
+    // Conversations
+    List<Conversation> allConversations = conversationStore.getAllConversations();
+    List<Message> messagesInConversation;
+
+    // HashMap to hold user IDs and the number of messages sent by that user
+    HashMap<UUID, Integer> userMessageCounts = new HashMap<UUID, Integer>();
+
+    int totalMessageCount = 0; // total number of messages sent by all users
+    int totalWordCount = 0; // total number of words sent by all users
+
+    for (Conversation c: allConversations) {
+
+      messagesInConversation = messageStore.getMessagesInConversation(c.getId());
+      totalMessageCount += messagesInConversation.size();
+
+      for (Message m: messagesInConversation) {
+
+        int previousMessageCount = userMessageCounts.getOrDefault(m.getAuthorId(), 0);
+        userMessageCounts.put(m.getAuthorId(), previousMessageCount + 1);
 
-    @Override
-    public void init() throws ServletException {
-        super.init();
+        totalWordCount += m.getContent().split("\\s+").length;
+      }
 
-        setAdminUsernames();
     }
 
-    void setAdminUsernames() {
-        admins = new HashSet<String>();
-        admins.add("daniel");
-        admins.add("leslie");
-        admins.add("serena");
-        admins.add("shana");
-        admins.add("kyra");
+    // compute most active user
+    Map.Entry<UUID, Integer> maxMessageCountUser = null;
+
+    for (Map.Entry<UUID, Integer> entry: userMessageCounts.entrySet()) {
+
+      if (maxMessageCountUser == null || entry.getValue() > maxMessageCountUser.getValue()) {
+        maxMessageCountUser = entry;
+      }
+
+    }
+    if (maxMessageCountUser != null && userStore.getUser(maxMessageCountUser.getKey()) != null) {
+      mostActiveUser = userStore.getUser(maxMessageCountUser.getKey()).getName();
+    }
+
+    String avgMessagesPerConvo;
+    if (allConversations.size() != 0) {
+      avgMessagesPerConvo = String.format("%.3f", (double) totalMessageCount / allConversations.size());
+    } else {
+      avgMessagesPerConvo = "0.000";
+    }
+
+    String avgWordsPerMessage;
+    if (totalMessageCount != 0) {
+      avgWordsPerMessage = String.format("%.3f", (double) totalWordCount / totalMessageCount);
+    } else {
+      avgWordsPerMessage = "0.000";
+    }
+
+    // Set attributes
+    request.setAttribute("numberOfUsers", allUsers.size());
+    request.setAttribute("newestUser", newestUser);
+    request.setAttribute("mostActiveUser", mostActiveUser);
+
+    request.setAttribute("numberOfConversations", allConversations.size());
+    request.setAttribute("numberOfMessages", totalMessageCount);
+    request.setAttribute("avgMessagesPerConvo", avgMessagesPerConvo);
+    request.setAttribute("avgWordsPerMessage", avgWordsPerMessage);
+
+  }
+
+  void addTestData() {
```

Comment:
Wow, this is awesome! (I was just thinking of adding some users manually through the UI...) It's so awesome, in fact, that I think it deserves its own PR, tests, and maybe a separate class. It would be nice to make this behavior available to admin users, maybe with some customization options (how many items to add, for example).

In the meantime, can you remove this from the current PR? The commit itself is saved if you want to reference it later: https://github.com/Sereezy/CodeU-Spring-2018/commit/d942efc47805f81bd7d2ea7ee4078371d015eee0